### ISSUE TITLE
The addon freezes Blender when it tries to load it.

### ISSUE THREAD

**Issue by inkoalawetrust** [abrasic/blendpresence#12]
I'm not sure how this is happening, but ever since today, the addon has caused my Blender installation to freeze. It causes Blender to hang when it tries to load it on startup, or if you try enabling it on the preferences window. Even trying to open my previous versions of 2.93 and 3.2 caused the same problem. The only way to stop the freezing was for me to remove it from the scripts folder so Blender can't find it and load it while it's using my main config.

I have no idea why that is happening, or why it spontaneously begun just today. But it did make me think that somehow my userpref.blend file was retroactively corrupted.

Here is the script of the addon from the appdata scripts folder for 3.3, along with my preferences file too. In case that can help you troubleshoot this bizarre bug.

[Prefs and addon.zip](https://github.com/abrasic/blendpresence/files/9639792/Prefs.and.addon.zip)

**Comment by abrasic** [comment]
Hi, thanks for your report and I apologize for the issue you've been facing. Unfortunately, I have not been able to reproduce it so far.

From the zip you provided, it appears you aren't using the latest version (you are using 1.7.1, the latest is [1.8.0](https://github.com/abrasic/blendpresence/releases/tag/1.8.0)). It's also important to note that an active internet connection is required in order for the add-on to properly make calls to the RPC, otherwise blender will hang until a connection is made (this also includes on startup). 

Make sure you're properly able to connect to [discord.com](discord.com), and ensure that no firewalls are somehow blocking in/out traffic from the Discord client and Blender.

**Comment by inkoalawetrust** [comment]
Installing the latest version seems to not cause Blender to freeze, also keep in mind that this just happened completely out of nowhere, and had never occurred before.

> From the zip you provided, it appears you aren't using the latest version (you are using 1.7.1, the latest is [1.8.0](https://github.com/abrasic/blendpresence/releases/tag/1.8.0)). It's also important to note that an active internet connection is required in order for the add-on to properly make calls to the RPC, otherwise blender will hang until a connection is made (this also includes on startup).
> 
> Perhaps you've been having connection issues lately? Make sure you're properly able to connect to [discord.com](https://discord.com), and ensure that no firewalls are somehow blocking in/out traffic from the Discord client and Blender.

Nah, I was not having any connection issues when that begun, especially with Discord, since I was actively asking people on Discord if they have any idea what could be causing this sudden freezing.

That being said, the addon _freezing the entire program if there is no internet connection_ sounds like a massive bug in itself. What if somebody is in a location without reliable internet, and their connection begins cutting off while they are working on a project ?

As for my GPU, it's a Gigabyte RTX 2060.

**Comment by abrasic** [comment]
> "the addon freezing the entire program if there is no internet connection sounds like a massive bug in itself"

Yes, obviously that is not the desired effect, but its a limitation with regular code execution, it freezes the entire UI until code completion. Ive had attempts of preventing load when no internet is detected to no avail, but I plan on reading up how to either use asyncio or using a modal operator so these problems dont happen in the future, however free time hasnt been on my side lately.

Before enabling the addon, could you try toggling the console via `Window > Toggle System Console` and see if there are any errors that may appear when you try to enable it?

Thanks

**Comment by inkoalawetrust** [comment]
Nothing weird is being logged on the console when enabling the addon, even when I reinstall the old 1.7.1 version and re-enable it.

```
reloading addon: blendpresence 1664131923.744588 1664131928.8932683 'C:\\Users\\inkoalawetrust\\AppData\\Roaming\\Blender Foundation\\Blender\\3.3\\scripts\\addons\\blendpresence\\__init__.py'
Modules Installed () from 'C:\\Users\\inkoalawetrust\\Documents\\Blender Plugins\\blendpresence1.7.1.zip' into 'C:\\Users\\inkoalawetrust\\AppData\\Roaming\\Blender Foundation\\Blender\\3.3\\scripts\\addons'
module changed on disk: 'C:\\Users\\inkoalawetrust\\AppData\\Roaming\\Blender Foundation\\Blender\\3.3\\scripts\\addons\\blendpresence\\__init__.py' reloading...
[BP] Connected!
```

**Comment by abrasic** [comment]
I have been able to replicate a freezing issue on latest software some while ago, however this was a time where Discord services were slow, thus causing the presence to freeze the client until the Discord RPC responded.

While most freezes end up to be an issue on pypresence's end, Most of the freezing problems will most likely be fixed if (a) I get the time to make the functions asynchronous and (b) I write better error handling; both of which are planned for the next release.

**Comment by ScrubN** [comment]
As of recent, this plugin seems to inconsistently cause Blender to completely lock up during plugin init.

I tested Blender versions 3.2-3.6.1 using plugin version 1.7.1, all of which exhibit the lock up. By launching Blender via a terminal, I was able to abort loading the plugin with `ctrl` + `c` showing the culprit is the RPC connector.
![image](https://github.com/abrasic/blendpresence/assets/72096833/10b6aff4-9fe0-4d01-8516-ffaef0c067bd)

> While most freezes end up to be an issue on pypresence's end

Based on the stacktrace it seems this is the case for me. For some stupid reason pypresence does not have a maximum iteration count when trying to connect to discord.
![image](https://github.com/abrasic/blendpresence/assets/72096833/ad967a4e-7324-4270-91cc-056b466c1769)
![image](https://github.com/abrasic/blendpresence/assets/72096833/b2937580-38d1-4d3d-a03d-e74159b1e277)
